code/src/components/ThoughtForm.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react'
+
+
+// Creating ThoughtForm for user to post new thoughts
+const ThoughtForm = (props) => {
+
+    const [newThought, setNewThought] = useState('')
+
+
+    // Prevents default for form
+    const handleFormSubmit = (event) => {
+    event.preventDefault()
+
+
+    fetch("https://happy-thoughts-technigo.herokuapp.com/thoughts", {
+        method: "POST",
+        headers: {'Content-Type': "application/json"},
+        body: JSON.stringify({ message: newThought })
+})
+    .then((res) => res.json())
+    .then((newThought) => props.setThoughts((thoughts) => [newThought, ...thoughts]))
+    .finally(() => setNewThought(''))
+}
+
+    return (
+
+    <div className="container">
+        <form onSubmit={handleFormSubmit}>
+            
+        
+            <div className="thought-form">
+                <h2>
+                    <label htmlFor="thoughtInput">What's making you happy right now?</label></h2>
+
+
+                <textarea
+                id="thoughtInput"
+                 className="text-input"
+                 rows='5'
+                 columns='140'
+                 placeholder="Type something... :)"
+                 type="text"
+                 name="thought"
+                 value={newThought}
+                 onChange={event => setNewThought(event.target.value)}
+                 >
+
+                </textarea>
+
+            <div className="input-container-details">
+
+                <div>
+                    <button disabled={newThought.length < 6 || newThought.length > 140 ? true : false} /* If input is less than 6 or longer than 140 characters, display button will be disabled */
```

Comment:
nice comment and script! so advanced! I didn't know there is an attribute for disabled option in button element!